scvi/dataloaders/_ann_dataloader.py

```diff
@@ -66,21 +88,32 @@ def __init__(
 
         # This is a lazy subset, it just remaps indices
         self.dataset = Subset(self._full_dataset, indices=self.indices)
-        sampler_cls = SequentialSampler if not shuffle else RandomSampler
-        sampler = BatchSampler(
-            sampler=sampler_cls(self.dataset),
-            batch_size=batch_size,
-            drop_last=drop_last,
-        )
+
+        if sampler is None:
+            sampler_cls = SequentialSampler if not shuffle else RandomSampler
+            sampler = BatchSampler(
+                sampler=sampler_cls(self.dataset),
+                batch_size=batch_size,
+                drop_last=drop_last,
+            )
+        if isinstance(sampler, BatchSampler):
+            batch_size = None  # disables PyTorch automatic batching
+            shuffle = False
+            drop_last = False
```

Comment:
what is going on here?

Reply:
We need to force these kwargs if the custom sampler they are passing in is a batch sampler, otherwise torch throws an error